internal/gmtcli/info.go

```diff
@@ -139,7 +165,169 @@ func setModemOption(c *cli.Context) error {
 	return nil
 }
 
-func printNodes(nodes []*gomeshproto.FromRadio_NodeInfo) {
+type MeshNode struct {
+	ID                  string  `json:"node_id,omitempty"`
+	NodeNumber          int     `json:"node_number,omitempty"`
+	LongName            string  `json:"longname,omitempty"`
+	ShortName           string  `json:"shortname,omitempty"`
+	AltitudeMeters      int     `json:"altitude_meters,omitempty"`
+	LatitudeDegrees     float64 `json:"latitude_degrees,omitempty"`
+	LongitudeDegrees    float64 `json:"longitude_degrees,omitempty"`
+	MACAddress          string  `json:"mac_address,omitempty"`
+	IsLicensed          bool    `json:"is_licensed"`
+	Role                string  `json:"role,omitempty"`
+	HardwareModel       string  `json:"hardware_model,omitempty"`
+	BatteryLevel        int     `json:"battery_level_percent,omitempty"`
+	Voltage             float64 `json:"battery_volts,omitempty"`
+	ChannelUtilization  float64 `json:"channel_utilization_percentage,omitempty"`
+	AirUtilTx           float64 `json:"air_utilization_transmit_percentage,omitempty"`
+	LastHeard           string  `json:"last_heard,omitempty"`
+	LastHeardSecondsAgo int     `json:"last_heard_seconds_ago,omitempty"`
+	LastHeardHuman      string  `json:"last_heard_human,omitempty"`
+}
+
+func ProtoNodeToMeshNode(node *gomeshproto.FromRadio_NodeInfo) MeshNode {
+	meshNode := MeshNode{}
+
+	if node.NodeInfo != nil {
+		meshNode.NodeNumber = int(node.NodeInfo.Num)
+	}
+
+	if node.NodeInfo.LastHeard != 0 {
+		lastSeen := time.Unix(int64(node.NodeInfo.LastHeard), 0).UTC()
+		meshNode.LastHeard = lastSeen.Format(time.RFC3339)
+
+		//calculate seconds ago
+		secondsAgo := time.Now().UTC().Sub(lastSeen).Seconds()
+		meshNode.LastHeardSecondsAgo = int(secondsAgo)
+		meshNode.LastHeardHuman = fmt.Sprintf(
+			"%s",
+			time.Since(lastSeen).Round(time.Second).String(),
+		)
+	}
+
+	if node.NodeInfo.User == nil {
+		return meshNode
+	}
+	if node.NodeInfo.User.LongName != "" {
+		meshNode.LongName = fmt.Sprintf(
+			"%s",
+			node.NodeInfo.User.LongName,
+		)
+	}
+
+	if node.NodeInfo.User.Id != "" {
+		meshNode.ID = fmt.Sprintf(
+			"%s",
+			node.NodeInfo.User.Id,
+		)
+	}
+
+	if node.NodeInfo.User.ShortName != "" {
+		meshNode.ShortName = fmt.Sprintf(
+			"%s",
+			node.NodeInfo.User.ShortName,
+		)
+	}
+
+	if node.NodeInfo.User.Macaddr != nil {
+		meshNode.MACAddress = fmt.Sprintf(
+			//format as 48 bit mac address in hex
+			"%02x:%02x:%02x:%02x:%02x:%02x",
```

Comment:
These changes to make things more easy to read are great! I don't _think_ they're getting applied to the non JSON view though. Could you make sure they get applied in both places?